sormas-app/app/src/main/java/de/symeda/sormas/app/lbds/LbdsDtoHelper.java

```diff
@@ -0,0 +1,206 @@
+		CaseDataDto.SHARED_TO_COUNTRY,
+		CaseDataDto.QUARANTINE_ORDERED_VERBALLY,
+		CaseDataDto.QUARANTINE_ORDERED_OFFICIAL_DOCUMENT,
+		CaseDataDto.QUARANTINE_EXTENDED,
+		CaseDataDto.QUARANTINE_REDUCED,
+		CaseDataDto.QUARANTINE_OFFICIAL_ORDER_SENT,
+		CaseDataDto.OVERWRITE_FOLLOW_UP_UNTIL,
+		"ownershipHandedOver");
+
+	private static final Map<String, Object> CASE_DEFAULT_VALUES = new HashMap<>();
+
+	private static Map<String, List<String>> LBDS_DTO_PROPERTIES = new HashMap<>();
+
+	static {
+		LBDS_DTO_PROPERTIES.put(PersonDto.class.getSimpleName(), LBDS_PROPERTIES_PERSON_DTO);
+		LBDS_DTO_PROPERTIES.put(CaseDataDto.class.getSimpleName(), LBDS_PROPERTIES_CASE_DATA_DTO);
+
+		CASE_DEFAULT_VALUES.put(CaseDataDto.INVESTIGATION_STATUS, InvestigationStatus.PENDING);
+		CASE_DEFAULT_VALUES.put(CaseDataDto.CASE_CLASSIFICATION, CaseClassification.NOT_CLASSIFIED);
+		CASE_DEFAULT_VALUES.put(CaseDataDto.CASE_CLASSIFICATION, InvestigationStatus.PENDING);
+		CASE_DEFAULT_VALUES.put(CaseDataDto.OUTCOME, CaseOutcome.NO_OUTCOME);
+		CASE_DEFAULT_VALUES.put(CaseDataDto.CASE_ORIGIN, CaseOrigin.IN_COUNTRY);
+		CASE_DEFAULT_VALUES.put(CaseDataDto.FOLLOW_UP_STATUS, FollowUpStatus.NO_FOLLOW_UP);
+	}
+
+	public static void stripLbdsDto(EntityDto entityDto) throws IntrospectionException, InvocationTargetException, IllegalAccessException {
+
+		String entityDtoClass = entityDto.getClass().getSimpleName();
+		List<String> lbdsProperties = LBDS_DTO_PROPERTIES.get(entityDtoClass);
+		if (lbdsProperties != null) {
+			for (PropertyDescriptor property : Introspector.getBeanInfo(entityDto.getClass()).getPropertyDescriptors()) {
+				String propertyName = property.getName();
+				boolean isLbdsProperty = LBDS_PROPERTIES_ENTITY_DTO.contains(propertyName) || lbdsProperties.contains(propertyName);
+				if (!isLbdsProperty) {
+					Method writeMethod = property.getWriteMethod();
+					if (writeMethod != null && writeMethod.getGenericParameterTypes().length == 1)
+						writeMethod.invoke(
+							entityDto,
+							new Object[] {
+								null });
+				}
+			}
+		} else {
+			throw new IllegalArgumentException(entityDtoClass + " is not intended for LBDS sending");
+		}
+	}
+
+	public static boolean isModifiedLbds(Person person, PersonDto personDto, boolean checkNonLbdsProperties)
+		throws IntrospectionException, InvocationTargetException, IllegalAccessException {
+		return isModifiedLbds(person, personDto, new PersonDtoHelper(), checkNonLbdsProperties);
+	}
+
+	public static boolean isModifiedLbds(Case caze, CaseDataDto caseDataDto, boolean checkNonLbdsProperties)
+		throws IntrospectionException, InvocationTargetException, IllegalAccessException {
+		return isModifiedLbds(caze, caseDataDto, new CaseDtoHelper(), checkNonLbdsProperties);
+	}
+
+	private static <ADO extends AbstractDomainObject, DTO extends EntityDto> boolean isModifiedLbds(
+		ADO ado,
+		DTO dto,
+		AdoDtoHelper<ADO, DTO> adoDtoHelper,
+		boolean checkNonLbdsProperties)
```

Comment:
WHen is this needed? Seem to contradict the purpose of the method.

Reply:
This is used in two places:
1. to determine if an Ado is suitable to be sent via LBDS (`checkNonLbdsProperties == false`, i.e. LBDS properties have been modified)
2. to check whether a Dto as sent via LBDS is equal to a local Ado, checked on receiving an LBDS response (`checkNonLbdsProperties == true`, the Ado has no extra properties which have not been sent to the SORMAS server)

Reply:
Obsolete after refactoring. Will be removed.